scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/FileSystemService.java

```diff
@@ -28,69 +27,72 @@ public FileSystemService() {
     }
 
     /**
-     * Creates the dataset (and its required directories) on the filesystem.
+     * Creates directories on the filesystem.
      *
-     * @param id       The id of the dataset.
-     * @param username The username of the dataset.
-     * @throws DatasetStorageException The exception thrown if the directory could not be created.
+     * @param username The username to be used as a directory name.
+     * @param type     The directory type (e.g., Dataset, etc.)
+     * @param id       The id to be used as a directory name.
+     * @throws FileSystemStorageException The exception thrown if the directory could not be created.
      */
     @Override
-    public void addDatasetDirectories(String id, String username) throws DatasetStorageException {
+    public void addDirectories(String username, String type, String id) throws FileSystemStorageException {
         // First, the correct directory must be created
-        Path path = createPath(username, id);
+        Path path = createPath(username, type, id);
         File directory = path.toFile();
         // if the Path was created successfully, create the directory
         try {
             if (!directory.exists()) {
                 Files.createDirectories(path);
             }
         } catch (IOException e) {
-            throw new DatasetStorageException("IOException occurred when attempting to add the Dataset", e);
+            throw new FileSystemStorageException("IOException occurred when attempting to add the Dataset", e);
         } catch (UnsupportedOperationException e) {
-            throw new DatasetStorageException("UnsupportedOperationException occurred when attempting to add the Dataset", e);
+            throw new FileSystemStorageException("UnsupportedOperationException occurred when attempting to add the Dataset", e);
         } catch (SecurityException e) {
-            throw new DatasetStorageException("SecurityException occurred when attempting to add the Dataset", e);
+            throw new FileSystemStorageException("SecurityException occurred when attempting to add the Dataset", e);
         }
     }
 
     /**
-     * A helper method to create a Path given a username and dataset name.
+     * A helper method to create a Path given a username and unique id.
      *
-     * @param username  The username of the owner of the dataset.
-     * @param datasetId The id of the dataset.
+     * @param username The username of the owner of the dataset.
+     * @param type     The directory type (e.g., Dataset, etc.)
+     * @param id       The id name.
```

Comment:
remove `name`? Also, this method will still work if id is `null`, right? The javadoc should say that. And last, you probably want to move the javadoc to the interface otherwise the autocomplete menu in Eclipse (and I presume IntelliJ as well) won't show it when the interface is used.

Reply:
Does the id parameter require a @Nullable annotation?
Putting the javadoc in the interface makes sense.  Along those lines, should I move the javadoc for each service we have to each interface we have?

Reply:
> Does the id parameter require a @Nullable annotation?
The javadoc should just say that id can be null and what happens if that is the case. If I do an autocomplete on fileSystemService, trying to choose a method I want to know if the method does what I need (e.g. just give me the path to the user directory if id is null). Does that make sense?

> Putting the javadoc in the interface makes sense. Along those lines, should I move the javadoc for each service we have to each interface we have?
Yes, ideally. I mention it here, since that was one of the method where the javadoc is really needed as I have several options how to call it.